cli/get_vhd_vsize.ml

```diff
@@ -14,16 +14,19 @@ let get_vhd_vsize filename =
     | End -> return ()
     | Cons (hd, tl) ->
       begin match hd with
-      | Fragment.Footer x ->
-	let size = x.Footer.current_size in
-	Printf.printf "%Ld\n" size;
-	exit 0
-      | _ ->
-	()
+        | Fragment.Footer x ->
+          let size = x.Footer.current_size in
+          Printf.printf "%Ld\n" size;
+          exit 0
+        | _ ->
+          ()
       end;
       tl () >>= fun x ->
-      loop x in
-  openstream (Input.of_fd (Vhd_format_lwt.IO.to_file_descr fd)) >>= fun stream ->
+      loop x
+  in
+  Vhd_format_lwt.IO.get_file_size filename >>= fun file_size ->
```

Comment:
I was going to suggest to use `Lwt_unix.LargeFile.fstat fd` instead, which would ensure we get the size of the same file even if it gets renamed, however that probably doesn't work right on block devices, and the above function does, so this is probably the best we can do.
I don't think we would intentionally rename or delete the source while `sparse-dd` is running, so this is probably fine.